### ISSUE TITLE
Additional properties true or false?

### ISSUE THREAD

**Issue by jschlyter** [ehn-digital-green-development/ehn-dgc-schema#19]
Do we allow the schema to be extended or is data per this schema locked to this, and only this.
If extensions are not allowed we should set additionalProperties to false.

**Comment by martin-lindstrom** [comment]
As I understand it additionalProperties=false should be added. Right @gabywh ?

**Comment by gabywh** [comment]
Discussed already with Semantic SG. My position: Postel's law. If there is more data in any JSON document then we simply ignore it. Applying additionalProperties=false would violate this so I removed it consistently.

**Comment by martin-lindstrom** [comment]
@gabywh Yes, but in that case "minimum data set" equals "maximum data set" doesn't hold anymore.

**Comment by jschlyter** [comment]
Perhaps we should set additionalProperties=True where we allow it to be explicit?

**Comment by gabywh** [comment]
@martin-lindstrom yes, I know.

Let me perhaps phrase it differently and somewhat more verbosely:
So the general approach started out with being very strict and enforcing "to the max" all the constraints we could possibly apply. 3 reasons for that (1) to catch outliers in the data and force at that point implementers to "clean up their (data) act" if they need to, and (2) to demonstrate clear compliance to both the EU proposed legislation as well as the eHealthNetwork documentation, (3) last and definitely not least: to help implementers in generating very clean and tight code, direct from spec and thus allow good interoperablility (Rose's Law, we'll come to Postel's law in a minute...) since the specification would leave very little - if any - room for manoeuvre (which is the bane of most developer's lives when having to deal with medical data in the first place).
However, the general approach has now shifted to one of being far more relaxed about all kinds of constraints (Postel's law) and accepting pretty much most things providing we can make some kind of sense of them (discussion on-going e.g. about the person name parts and what is mandatory and what is not).

My own personal opinion: in this case, I think applying Postel's law here makes for a very sloppy spec and whilst I in general agree with Postel's law for interoperability of already-existing systems, I think Rose's law makes a very valid point that applying Postel's law without constraint leads to very sloppy systems -> entropy, if you so will.  So I would prefer a much tighter, cleaner JSON schema here -> including additionalProperties=False. However, it is not about my own personal opinion but what is generally required and whilst mu opinion may differ on a personal level, I still aim to follow the overall approach. This entails having a generally consistent approach throughout, which I think only helps (having a mix of strict and lax approaches at seemingly arbitrary points just leads to confusion). So that general approach is de-constraining the constraints, again without going so far as to say "everything is a string", which is too extreme. I hope to have found a balance here in that approach -> and that entailed for me removing any additionalProperties=false constraints.

@jschlyter I do like your suggestion of at least being explicit about intentions: in general, good programming practice. Yet JSON itself is designed to have  lot of implicits and be flexible rather than constraining (although I do find comments not supported in JSON in general to be rather irritating, but again that's a personal opinion ;). For JSON schema additionalProperties=True is "on" by default in any case, does being explicit here actually help or just add more parsing (also human parsing ;) "noise"? If we state additonalProperties=True then it would imply we set in other places additionalProperties=False elsewhere, at least that would be my expectation as a developer. If we allow the default of additionalProperties=True everywhere (as in the general, less-constraining approach), then there would be no need to specify it at all.

So in summary: yes, I did think about it, quite a bit in fact ;)

**Comment by jschlyter** [comment]
So how do we proceed? True/False/undef

**Comment by gabywh** [comment]
1. well it can never be undef -> is by default defined to be True
2. but I get what you mean: not explicitly defining it ;)
3. How to proceed? Leave as-is

**Comment by martin-lindstrom** [comment]
As say. Leave it "as is", with no mentioning of additionalProperties. In this way we don't explicitly invite for extensions, but does not actively prevent it either.

**Comment by martin-lindstrom** [comment]
I say: Leave it "as is". In that way we don't explicitly invite for extensions, but do not actively prevent it either. A compromise.
